bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigValueCreator.java

```diff
@@ -254,29 +256,32 @@ private Object getElementValue(Type elementType, TomlNode tomlValueNode) {
     }
 
     private BMap<BString, Object> createRecordValue(TomlNode tomlNode, Type type) {
-        RecordType recordType;
-        String recordName;
-        if (type.getTag() == TypeTags.RECORD_TYPE_TAG) {
-            recordName = type.getName();
-            recordType = (RecordType) type;
+        RecordType mutableType;
+        Optional<IntersectionType> intersectionType = ((IntersectableReferenceType) type).getIntersectionType();
+        if (intersectionType.isPresent()) {
```

Comment:
Shall we add a comment here. And create and issue to revert this once we fix the record init

Reply:
As per the spec issue conclusion (https://github.com/ballerina-platform/ballerina-spec/issues/544), this is the suggested way of creating the intersection values. 
I'll add a comment including the value creation-related info.